source/GLaDOS.cpp

```diff
@@ -248,6 +296,53 @@ void GLaDOS::update(float dt) {
 		break;
 	}
 
+	//	for (int i = 0; i < maxButtons; i++) {
+	//		if (buttonFree.at(i)) {
+	//			float angle = (float)(rand() % (int)(ship->getSize()));
+	//			bool goodAngle = true;
+	//			for (int j = 0; j < ship->getDonuts().size(); j++) {
+	//				float diff =
+	//					ship->getSize() / 2 -
+	//					abs(abs(ship->getDonuts().at(j)->getAngle() - angle) - ship->getSize() / 2);
+	//				if (diff < 40) {
+	//					goodAngle = false;
+	//					break;
+	//				}
+	//			}
+	//			if (!goodAngle) {
+	//				continue;
+	//			}
+	//			float pairAngle;
+	//			int pairID;
+	//			float minButtonDiff = 100;
+	//			for (int j = 0; j < maxButtons; j++) {
+	//				if (i != j && buttonFree.at(j)) {
+	//					ship->getButtons().at(i)->setPair(ship->getButtons().at(j), j);
+	//					ship->getButtons().at(j)->setPair(ship->getButtons().at(i), i);
+	//					pairAngle = (float)(rand() % (int)(ship->getSize()));
+	//					while(abs(pairAngle - angle) < minButtonDiff) {
+	//						pairAngle = (float)(rand() % (int)(ship->getSize()));
+	//					}
+	//					ship->getButtons().at(j)->setAngle(pairAngle);
+	//					ship->getButtons().at(j)->clear();
+	//					buttonFree.at(j) = false;
+	//					pairAngle = ship->getButtons().at(j)->getAngle();
+	//					break;
+	//				}
+	//			}
+	//			if (ship->getButtons().at(i)->getPair() == nullptr) {
+	//				continue;
+	//			} else {
+	//				ship->getButtons().at(i)->setAngle(angle);
+	//				ship->getButtons().at(i)->clear();
+	//				buttonFree.at(i) = false;
+	//				pairID = ship->getButtons().at(i)->getPairID();
+	//				mib->createButtonTask(angle, i, pairAngle, pairID);
+	//				break;
+	//			}
+	//		}
+	//	}
+
```

Comment:
Delete commented code if it's not needed.

Reply:
kept it in case i did the level stuff incorrectly and needed to go back to original before i added level support. but i can delete it if level support looks good